lib/attn.rb

```diff
@@ -1,16 +1,74 @@
-def attn(method = nil, color = 'yellow')
-	colors = {'red' => '031','green' => '032','yellow' => '033','blue' => '034','magenta' => '035','cyan' => '036'}
-
-	color,method = method,nil if colors.keys.include?(method)
-	color = 'yellow' unless colors.keys.include?(color)
-	
-	unless method
-		print "\033[#{colors[color]}m* * * * * * * * * * * * * * * * * * * * * * * * *\033[0m\n"
+require 'colorize'
+
+$COLORS = ['red','green','yellow','blue','magenta','cyan']
+
+def attn(variable = nil, color = '')
+
+	color,variable = variable,nil if $COLORS.include?(variable)
+	color = $COLORS.sample unless $COLORS.include?(color)
+
+	unless variable
+		face_break
 	else
-		proc = Proc.new { method }
-		print "\033[#{colors[color]}m* * * * * * * * * * START * * * * * * * * * * * *\033[0m\n"
-		p proc.call
-		print "\033[#{colors[color]}m* * * * * * * * * * FINISH  * * * * * * * * * * *\033[0m\n"
+		@color,@variable = color,variable
+		detail_report
+		display_variable
+		close_report
+	end
+
+end
+
+private
+
+def display_variable
+	proc = Proc.new { @variable }
+	puts
+	p proc.call
+	puts
+end
+
+def face_break
+	chars = %w"😎 😇 😈 👹 👺 👻 👿 💀 👽 😂 🤣"
+	puts "#{chars.sample}  " * 40
+end
+
+def close_report
+	str = "⬆ " * 15
+	str += " FINISH "
+	str += "⬆ " * 15
+	puts str.send(@color)
+end
+
+def detail_report
+	secondary_color = @color == 'red' ? 'blue' : 'red'
+	str = "\n"
+	str += "⬇ ".send(@color) * 5
```

Comment:
Do you think there's any value in creating the down-arrow string before doing the .send, and then .send-ing only with the final result?